massa-execution-worker/src/active_history.rs

```diff
@@ -167,25 +168,26 @@ impl ActiveHistory {
             })
             .collect()
     }
+    */
 
-    /// Traverse the whole history and return every deferred credit _at_ `slot`
+    /// Return deferred credits in active history _at_ `slot`
     ///
     /// # Arguments
     /// * `slot`: slot _at_ which we fetch the credits
     pub fn fetch_all_deferred_credits_at(&self, slot: &Slot) -> PreHashMap<Address, Amount> {
         self.0
             .iter()
-            .filter_map(|output| {
-                output
+            .rev()
+            .find_map(|exc_out| {
+                exc_out
```

Comment:
Problem: if at slot 3 we register a deferred credit targeting slot 6, and at slot 4 we register another deferred credit targeting slot 6 but for another address, then when we run this it will only return the second one

Reply:
I gueds we can retain the old version for this one

Reply:
Note: fetch_all_deferred_credits_at has the same problem !

Reply:
Fixed